### PR TITLE
Add `Auto` option for `IptablesBackend` to FelixConfiguration

### PR DESCRIPTION
**Pull request by huiyizzz**
## Description

We support `auto` to be set in IptablesBackend as an env var in Felix. But for the FelixConfiguration API resource, there is no `Auto` option. In the API code we say unset means legacy, but actually it is auto.

This PR adds an "Auto" option for `IptablesBackend` and updates the comments to say "Auto" is the default.
## Related issues/PRs

https://github.com/projectcalico/calico/issues/6772
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Add missing Auto option for IptablesBackend FelixConfiguration field
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Review by caseydavenport — CHANGES_REQUESTED** [review]
I'm not sure yet if this is the right fix - we do support auto-detection of the iptables backend in Felix. Certainly the changes to config_params.go seem wrong to me.

I think this might just be a case of the documentation being confusing.

i.e.,

- We support a value of Auto being set as an env var in FELIX_IPTABLESBACKEND
- We do not support a value of Auto being set in the v3 FelixConfiguration API.

This is technically OK since they are different APIs, but our docs imply that the same values are appropriate on both the env var and v3 APIs.

File: api/pkg/apis/projectcalico/v3/felixconfig.go [hunk 1]
```diff
@@ -133,7 +134,7 @@ type FelixConfigurationSpec struct {
 	// disable iptables refresh. [Default: 90s]
 	IpsetsRefreshInterval *metav1.Duration `json:"ipsetsRefreshInterval,omitempty" configv1timescale:"seconds"`
 	MaxIpsetSize          *int             `json:"maxIpsetSize,omitempty"`
-	// IptablesBackend specifies which backend of iptables will be used. The default is legacy.
+	// IptablesBackend specifies which backend of iptables will be used. The default is auto.
```
Comment (caseydavenport) [inline api/pkg/apis/projectcalico/v3/felixconfig.go:137]: This should be "Auto" (capitalized, to match the enum value).

**Review by caseydavenport — CHANGES_REQUESTED** [review]
I think we're missing a couple of things in this PR:

- Docs updates to this file: https://github.com/projectcalico/calico/blob/master/calico/reference/resources/felixconfig.md
- Unit tests to assert that Auto is accepted through validation, and that Felix config parsing handles it correctly.

I'd look at these two suites:

- https://github.com/projectcalico/calico/blob/master/felix/config/config_params_test.go#L136
- https://github.com/projectcalico/calico/blob/master/libcalico-go/lib/validator/v3/validator_test.go

File: libcalico-go/lib/validator/v3/validator_test.go [hunk 2]
```diff
@@ -665,6 +667,7 @@ func init() {
 		Entry("should reject : in an interface", libapiv3.WorkloadEndpointSpec{InterfaceName: "Invalid:Intface"}, false),
 
 		// (API) FelixConfiguration.
+		Entry("should accept a valid IptablesBackend value 'Auto'", api.FelixConfigurationSpec{IptablesBackend: &iptablesBackendAuto}, true),
```
Comment (coutinhop) [inline libcalico-go/lib/validator/v3/validator_test.go:670]: Could you add cases for accepting "NFT" and "Legacy" and rejecting an invalid value here as well?

**Review by mgleung — APPROVED** [review]

**Pull request merged**